src/main/java/com/faforever/client/mod/ModService.java

```diff
@@ -307,45 +308,63 @@ public void overrideActivatedMods(Collection<ModVersionBean> modVersions) {
   private Set<String> readActiveMods() throws IOException {
     Path preferencesFile = preferencesService.getPreferences().getForgedAlliance().getPreferencesFile();
     Set<String> activeMods = new HashSet<>();
-
-    String preferencesContent = Files.readString(preferencesFile, StandardCharsets.UTF_8);
-    Matcher matcher = ACTIVE_MODS_PATTERN.matcher(preferencesContent);
-    if (matcher.find()) {
-      Matcher activeModMatcher = ACTIVE_MOD_PATTERN.matcher(matcher.group(0));
-      while (activeModMatcher.find()) {
-        String modUid = activeModMatcher.group(1);
-
-        if (Boolean.parseBoolean(activeModMatcher.group(2))) {
-          activeMods.add(modUid);
+    List<Charset> preferredEncodings = new ArrayList<>(List.of());
+    Map<String, Charset> map = Charset.availableCharsets();
+    for (Map.Entry<String, Charset> entry : map.entrySet()) {
+      Charset charset = entry.getValue();
+      preferredEncodings.add(charset);
+    }
+    for (Charset charset : preferredEncodings) {
+      try {
+        String preferencesContent = Files.readString(preferencesFile, charset);
+        Matcher matcher = ACTIVE_MODS_PATTERN.matcher(preferencesContent);
+        if (matcher.find()) {
+          Matcher activeModMatcher = ACTIVE_MOD_PATTERN.matcher(matcher.group(0));
+          while (activeModMatcher.find()) {
+            String modUid = activeModMatcher.group(1);
+            if (Boolean.parseBoolean(activeModMatcher.group(2))) {
+              activeMods.add(modUid);
+            }
+          }
         }
+        break;
+      } catch (MalformedInputException e) {
+        throw new AssetLoadException("Could not read mod state", e, "mod.errorReadingMods");
       }
     }
```

Comment:
Right now this try catch loop still throws an exception rather than moving onto the next encoding. It would be better to just have the extraction of the preferencesContent in the for loop with the try catch and then throw an assetloadexception if the preferencesContent is null. So more like below
```suggestion
      try {
        String preferencesContent = Files.readString(preferencesFile, charset);
        break;
      } catch (MalformedInputException e) {
        log.warn(message);
      }
        
        Matcher matcher = ACTIVE_MODS_PATTERN.matcher(preferencesContent);
        if (matcher.find()) {
          Matcher activeModMatcher = ACTIVE_MOD_PATTERN.matcher(matcher.group(0));
          while (activeModMatcher.find()) {
            String modUid = activeModMatcher.group(1);
            if (Boolean.parseBoolean(activeModMatcher.group(2))) {
              activeMods.add(modUid);
            }
          }
        }
```